cloudinary-core/src/test/java/com/cloudinary/analytics/AnalyticsTest.java

```diff
@@ -0,0 +1,93 @@
+package com.cloudinary.analytics;
+
+import com.cloudinary.Cloudinary;
+import com.cloudinary.utils.Analytics;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+public class AnalyticsTest {
+
+    private Cloudinary cloudinary;
+
+    @Rule
+    public TestName currentTest = new TestName();
+
+    @Before
+    public void setUp() {
+        System.out.println("Running " + this.getClass().getName() + "." + currentTest.getMethodName());
+        this.cloudinary = new Cloudinary("cloudinary://a:b@test123?load_strategies=false");
+    }
+
+    @Test
+    public void testToQueryParam() {
+        Analytics analytics = new Analytics("F", "2.0.0", "1.8.0");
+        String result = analytics.toQueryParam();
+        Assert.assertEquals(result, "_a=AFAACMh0");
+    }
+
+    @Test
+    public void testUrlWithAnalytics() {
+        cloudinary.config.analytics = true;
+        cloudinary.setAnalytics(new Analytics("F", "2.0.0", "1.8.0"));
+        String url = cloudinary.url().generate("test");
+        Assert.assertEquals(url, "http://res.cloudinary.com/test123/image/upload/test?_a=AFAACMh0");
+        cloudinary.config.analytics = false;
```

Comment:
this (and all the other occurrences) should be set in the `tearDown` method. If the assertion fails, this code is never executed. This state might also affect other tests.